test/cdk-lib/bedrock/application-inference-profile.test.ts

```diff
@@ -0,0 +1,123 @@
+/**
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ *  with the License. A copy of the License is located at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
+ *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
+ *  and limitations under the License.
+ */
+import * as cdk from 'aws-cdk-lib';
+import { Match, Template } from 'aws-cdk-lib/assertions';
+import { CfnApplicationInferenceProfile } from 'aws-cdk-lib/aws-bedrock';
+import { AwsSolutionsChecks } from 'cdk-nag';
+import * as bedrock from '../../../src/cdk-lib/bedrock';
+
+describe('CDK-Created-Application-Inference-Profile', () => {
+  let stack: cdk.Stack;
+
+  beforeEach(() => {
+    const app = new cdk.App();
+    cdk.Aspects.of(app).add(new AwsSolutionsChecks());
+    stack = new cdk.Stack(app, 'TestStack');
+  });
+
+  test('Basic Creation with a foundation model', () => {
+    new bedrock.ApplicationInferenceProfile(stack, 'TestAIP', {
+      inferenceProfileName: 'TestAIP',
+      description: 'This is a test application inf profile',
+      modelSource: bedrock.BedrockFoundationModel.ANTHROPIC_CLAUDE_SONNET_V1_0.asArn(stack),
```

Comment:
I was thinking we probably can implement IModel for AWS-defined FMs without changing the interface using [PseudoParameters](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.Aws.html). This would remove weird syntax and simplify its usage. Usually this is not commonly used as this will simply translate to the equivalent [CFN PseudoParameter](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/pseudo-parameter-reference.html), and does not work well for most resources when referencing resources across stacks, however for FMs this would actually work.

Reply:
Done by you :)